ang/civicase/case/actions/directives/case-actions.directive.js

```diff
@@ -378,18 +101,38 @@
             _.remove($scope.caseActions, { action: 'changeStatus(cases)' });
           }
         }
+        refreshDataForActions();
       });
 
       /**
-       * Aligns the dialog box center to the screen
+       * Get Case Action Service
        *
-       * @param {jQuery} dialog box to be aligned center
+       * @param {String} action
+       * @return {Object/Boolean}
        */
-      function alignDialogBoxCenter (dialog) {
-        if (dialog && dialog.data('uiDialog')) {
-          dialog.parent().position({ 'my': 'center', 'at': 'center', 'of': window });
+      function getCaseActionService (action) {
+        try {
+          return $injector.get(action + 'CaseAction');
+        } catch (e) {
+          return false;
```

Comment:
why return `false` ? Just return `undefined`.

Also, if I'm defining an action and the service has not been created for the action I rather throw an error so the developer creates the service instead of failing silently.

Or is there a genuine case where we don't want to create the service for an action?

Reply:
Yes I agree, removed the try catch block.